### DISCUSSION TITLE
Xdrip+ not uploading data to Nightscout after switching to Libre 2 sensor

### DISCUSSION THREAD

**Discussion by agrmonitorcgm** [NightscoutFoundation/xDrip#3563] [Q&A]
Hey there. My son has had type 1 diabetes for more than 6 years now. For a long period of time, we happily used this flow: Libre 1 Sensor > Miao Miao 3 (Miao Miao 1 before that) > xDrip+ (on Android) > fly. io servers > Nightscout. For over 2 years, this chain of apps and services ran like clockwork.

Today, we decided to switch to Libre 2 sensors. I successfully installed the Libre 2 app to activate the sensor. After that, I installed the Juggluco app and set it up following guides. It works fine; glucose data has been sent for a few hours with no connection losses.

According to the guides, I updated my xDrip+ app to the latest version and switched the Hardware Data Source to Libre (patched app). xDrip+ received the data and is correctly receiving glucose data from Juggluco.

However, xDrip+ suddenly became unable to send data to Nightscout, even though I haven't changed anything in those settings, including my Nightscout URL. When I try to poll the data, it shows the same error again and again (see screenshot).

I also tried my nightscout URL in the Juggluco app and got an error as well. java.io.filenotfoundexception https://myurl.fly.dev/api/v1/entries
(have no idea we he added entries to my URL)

Could someone please help me solve this problem?
![2024-07-08 01_20_59-juggluco-or-xdrip-not-uploading-data-to-nightscout-after-v0-9vc6od0ok5bd1 webp  ](https://github.com/NightscoutFoundation/xDrip/assets/70815961/be40555d-eaeb-4af3-929f-b0be4db2afb4)

**Comment by Navid200** [comment]
Can you access your Nightscout site from a computer?

Can you access your Nightscout site in a browser on the same phone that you are running xDrip on?

**Reply by agrmonitorcgm** [reply]
Yes, I paste my nightscout URL in web browser on my pc with adding /entries to the end and it works fine, shows some numbers and so on. 
![image](https://github.com/NightscoutFoundation/xDrip/assets/70815961/14bb6f33-736e-42cb-af72-54a0b75ddfda)

If I add nothing to the end I get Cannot get /api/v1/
![image](https://github.com/NightscoutFoundation/xDrip/assets/70815961/a009a95f-de8a-439c-b9bb-97519bae572e)

The same behavior on the device with juggluco and nightscout.

**Reply by agrmonitorcgm** [reply]
May be it is related to the logs on my machines.

![image](https://github.com/NightscoutFoundation/xDrip/assets/70815961/58ac3071-bfa0-4ea2-81f6-b7b1b0a54a9e)

**Reply by PhilTU** [reply]
usually you can view a graphical nightscout output in a web browser without the api etc on the end for example https://myurl.fly.dev although this may not apply with higher security (and I'm not familair with that hosting)

**Reply by agrmonitorcgm** [reply]
Yes. And My URL worked correctly all the time. Turned out juggluco requires no "/api/v1/" addition in the URL field thought xdrip+ requires.

**Reply by Navid200** [reply]
I'm glad it is working.  The following is for anyone else who may find this later experiencing something similar.
<br/>    
  
If you experience a problem with xDrip uploading, the first thing we ask you is to show your Nightscout in a browser.
That's what you see as the very first image if you go to the Nightscout guide: https://nightscout.github.io/
Which is this:  
![Screenshot 2024-07-08 122325](https://github.com/NightscoutFoundation/xDrip/assets/51497406/ad1ea978-2570-4daf-9aba-69e085b27ec8)


If you can see that and xDrip does not upload to it, this explains what you need to do in xDrip so that it can:  
https://navid200.github.io/xDrip/docs/Nightscout-Settings.html  

If you can see your site in a browser on your uploader phone and if you have followed the instructions for setting up xDrip as an uploader and it does not work, please tell us and we will help you sort it out.  
  
If you post something other than the Nightscout curve as shown in the Nightscout guide, it will not help.

**Comment by agrmonitorcgm** [comment]
I've fixed this. For some reason my api_secret was incorrect, though I didn't touch it for years. it was like https://myapisecret.lalala. but my api_secret is myapisecretislonger. It fixed the problem right away.
